### ISSUE TITLE
BUG REPORT: NoClassDefFoundError: java/sql/Xxx

### ISSUE THREAD

**Issue by DragonKnightOfBreeze** [google/gson#1537]
Please consider using `try...catch` exp with code in that relates to `java.sql.Xxx`,
like `java.sql.Date`,`java.sql.Time`.
Otherwise, it will show `NoClassDefFoundError` sometimes.
Eg: use Gson by a kotlin script

```
//line 265 in Gson.java
//The following code should surround with try catch.

factories.add(TimeTypeAdapter.FACTORY);
factories.add(SqlDateTypeAdapter.FACTORY);
```

**Comment by DragonKnightOfBreeze** [comment]
`requires java.sql` in module-info.java seems do noting.

**Comment by Marcono1234** [comment]
Similar to / same as #1629

**Comment by Marcono1234** [comment]
Was probably fixed by #1707.
